### PR TITLE
Fix show stmt with predicate bug

### LINKED ISSUE #3708: Show variables where error in 2.1
### Steps to reproduce the behavior (Required)
```
mysql> show variables where Variable_name = "time_zone";
ERROR 1064 (HY000): Shouldn't reach here
mysql>
```

### StarRocks version (Required)
branch-2.1

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #3708 #3718

### CODE REVIEW

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowTableStatusStmt.java [hunk 1]
```diff
@@ -101,15 +103,10 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
...
-    public SelectStmt toSelectStmt(Analyzer analyzer) throws AnalysisException {
+    public QueryStatement toSelectStmt() throws AnalysisException {
         if (where == null) {
             return null;
         }
-        if (selectStmt != null) {
-            return selectStmt;
-        }
-
-        analyze(analyzer);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowTableStatusStmt.java:112]: analyze function check the access permissions，could just delete it ?

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowColumnStmt.java [hunk 2]
```diff
@@ -124,17 +125,14 @@ public void init() {
     }
 
     @Override
-    public SelectStmt toSelectStmt(Analyzer analyzer) throws AnalysisException {
+    public QueryStatement toSelectStmt() throws AnalysisException {
         if (where == null) {
             return null;
         }
-        if (selectStmt != null) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowColumnStmt.java:131]: selectStmt always be null now?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/analysis/ShowColumnStmt.java:131]: Yes, no special judgment is required, and a SelectStmt is stored. Show statements are all executed once

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowTableStatusStmt.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/analysis/ShowTableStatusStmt.java:112]: Under the new Analyzer system, the verification of permissions should be carried out in PrivilegeChecker, and there is no problem with deletion here. But unfortunately, we missed this point. I will submit another PR to resolve the statements currently supported by the new Parser and improve the code related to authorization authentication.

**Review by Youngwb — APPROVED** [review]

**Pull request merged**